app/models/invite.rb

```diff
@@ -160,6 +183,49 @@ def self.create_invite_by_email(email, invited_by, opts = nil)
     invite
   end
 
+  # multiple use invite link
+  def self.create_invite_link(invited_by, opts = nil)
+    opts ||= {}
+
+    group_ids = opts[:group_ids]
+    max_redemptions_allowed = opts[:max_redemptions_allowed].presence || 5
+    invite_link_expires_at = opts[:invite_link_expires_at].presence || 1.month.from_now
+
+    create_args = {
+      invited_by: invited_by,
+      max_redemptions_allowed: max_redemptions_allowed,
+      invite_link_expires_at: invite_link_expires_at,
+      emailed_status: emailed_status_types[:not_required]
+    }
+    invite = Invite.create!(create_args)
+
+    if group_ids.present?
+      group_ids = group_ids - invite.invited_groups.pluck(:group_id)
+
+      group_ids.each do |group_id|
+        invite.invited_groups.create!(group_id: group_id)
+      end
+    end
+
+    invite.reload
+    "#{Discourse.base_url}/invites/#{invite.invite_key}" if invite
+  end
+
+  # redeem multiple use invite link
+  def redeem_invite_link(email: nil, username: nil, name: nil, password: nil, user_custom_fields: nil, ip_address: nil)
+    if is_invite_link? && !expired? && !redeemed? && !destroyed? && link_valid?
+      lower_email = Email.downcase(email)
+      user = User.find_by_email(lower_email)
+      raise UserExists.new I18n.t("invite.invite_link_user_exists") if user.present?
+
+      self.update_column(:email, email)
```

Comment:
That has crossed my mind as well. What do you recommend here?

Reply:
I think `InviteRedeemer` will need updating so that it can be optionally be passed an email address. If it is supplied with an email address then it uses that instead of `invite.email`. 

Would that work?

Reply:
That sounds doable. Let me try implementing that.

Reply:
Okay, I've updated `InviteRedeemer` to accept email and made it so that we pass email whenever `InviteRedeemer` is initialized.